src/Microsoft.DotNet.ProjectModel/AnalyzerOptions.cs

```diff
@@ -25,13 +25,17 @@ public class AnalyzerOptions
 
         public override bool Equals(object obj)
         {
-            if (obj == null)
+            if (ReferenceEquals(obj, null))
             {
                 return false;
             }
+            if (ReferenceEquals(obj, this))
+            {
+                return true;
+            }
 
             var options = obj as AnalyzerOptions;
-            return obj != null && (this == options);
```

Comment:
Fixing the `options != null` part is a good change, but I think the second half should continue being `(this == options)`.  The reasoning being that the "rules" for what makes two AnalyzerOptions equal should be in a single spot.  Here the rule that the "LanguageIds should be the same" is duplicated into 2 spots.